components/zeekopenbsm/src/openbsmconsumer.cpp

```diff
@@ -0,0 +1,363 @@
+#include "openbsmconsumer.h"
+#include "openbsm_utils.h"
+
+#include <chrono>
+#include <condition_variable>
+#include <ctime>
+#include <future>
+#include <iomanip>
+#include <thread>
+
+namespace zeek {
+
+namespace {
+#define AUDIT_PIPE_PATH "/dev/auditpipe"
+} // namespace
+
+struct OpenbsmConsumer::PrivateData final {
+  PrivateData(IZeekLogger &logger_, IZeekConfiguration &configuration_)
+      : logger(logger_), configuration(configuration_) {}
+
+  IZeekLogger &logger;
+  IZeekConfiguration &configuration;
+
+  EventList event_list;
+  std::mutex event_list_mutex;
+  std::condition_variable event_list_cv;
+
+  std::atomic_bool terminate_producer{false};
+};
+
+OpenbsmConsumer::OpenbsmConsumer(IZeekLogger &logger,
+                                 IZeekConfiguration &configuration)
+    : d(new PrivateData(logger, configuration)) {
+
+  // start fetching events from pipe in a separate thread
+  std::thread{&OpenbsmConsumer::fetchEventsFromPipe, this}.detach();
```

Comment:
I did not use `join` because there was no reason for me to wait for consumer thread. As soon as the destructor for the producer is called, the consumer thread will be stopped. As of now, we don't have multiple Openbsm producers. We just have one Openbsm producer and one consumer; thus, leaking will never happen.

Reply:
Not waiting here can obscure any issues in the shutdown of the threads (which might be introduced inadvertently a some later time). Let's make sure the thread actually has terminated by joining it.

Reply:
Alright. That makes sense. I'll add `join` in the destructor.